homeassistant/components/freebox/binary_sensor.py

```diff
@@ -44,10 +48,111 @@ async def async_setup_entry(
         for description in RAID_SENSORS
     ]
 
+    for nodeid, node in router.home_devices.items():
+        if nodeid in tracked:
+            continue
+        if node["category"] == FreeboxHomeCategory.PIR:
+            binary_entities.append(FreeboxPirSensor(hass, router, node))
+        elif node["category"] == FreeboxHomeCategory.DWS:
+            binary_entities.append(FreeboxDwsSensor(hass, router, node))
+
+        sensor_cover_node = next(
+            filter(
+                lambda x: (x["name"] == "cover" and x["ep_type"] == "signal"),
+                node["show_endpoints"],
+            ),
+            None,
+        )
+        if sensor_cover_node and sensor_cover_node.get("value") is not None:
+            binary_entities.append(FreeboxCoverSensor(hass, router, node))
+
+        tracked.add(nodeid)
```

Comment:
As `sensor.py`

Reply:
sensor.py is for call, connections,disk,temp and battery sensors
binary_sensor.py is for PIR (infrared), DWS (switchs) and covers

Reply:
I know 😂

I am talking about the code logic as battery sensor PR #99222

Reply:
ok. Object's Type  added in list

Reply:
I was not talking about the typing but about:
- removing tracked sensors
- not using filter-next functions but a for-if https://github.com/home-assistant/core/commit/4c9e5442851ee853971f31b8888add06d435ced5

Reply:
VH

Reply:
ok. I get it. Done